slides/src/content/rxjs-observables.md

```diff
@@ -1,164 +1,191 @@
 
+Notes:
+Objectives:
+1. To be able to inject http into an Angular component, get a value from a typical REST endpoint and bind it directly to the template.
+2. To be able to explain what the `async` pipe does and describe how you would mimic the behaviour of the `async` pipe in a component by subscribing to an observable.
+
 ---
 
 ## What are Observables?
 
-* The `Observable` is a proposed standard for managing asynchronous data for ES7 and beyond.
-* Available for use via [RxJS library](https://github.com/ReactiveX/rxjs).
-* Allows for effective handling of a *stream* of asynchronous events with operations similar to 
-array methods like `map` and `filter`.
-* Similar in purpose to Promises but also more powerful with added features like disposability.
-* Angular uses it, a lot!
-
----
+* ECMA Stage 1, expected to move to Stage 2 soon.
+* Available for use now via [RxJS library](https://github.com/ReactiveX/rxjs).
+* Used in many languages and frameworks.
+* Used extensively in Angular.
+* Observables help with asynchronous behaviours.
 
-## Why Not Just Use Promises?
++++
 
-* Both `Promises` and `Observables` are used for dealing with asynchronous code. However, 
-Observables are considered more powerful because:
-  * `Observables` are more expressive.
-  * `Observables` are cancellable.
-  * `Observables` have rich API that make operations like retrying much easier (*i.e.*, with `retry` and 
-  `retryWhen` operators).
+## Quiz 1
 
----
+1. An observable will:
+  1. two-way data bind a property
+  2. push notifications to an observer
+  3. let angular know when change detetection is requried
 
-## Creating an Observable
++++
 
-```js
-import {Observable} from 'rxjs/Observable';
+## Definition from rxjs docs
 
-const observable = new Observable(observer => {
-  observer.next(1);
-  observer.next(2);
-  observer.complete();
-})
-```
-Creating the Observable alone does not trigger anything.
-Creating an Observable is like defining a function: 
-- A function does not do anything until it is *called*. 
-- Likewise, an Observable does not start streaming data until it's been *subscribed*.
+> The Observer and Observable interfaces provide a generalized mechanism for push-based notification, also known as the observer design pattern. The Observable object represents the object that sends notifications (the provider); the Observer object represents the class that receives them (the observer).
 
 ---
 
 ## Subscribing to an Observable
 
-```js
-const subscription = observable.subscribe(
-  value => console.log('value', value),    // handles next
-  error => console.log('error', error),    // handles error 
-  () => console.log('complete'),  // handles complete
-);
-```
+Here is how we tell an observable we want to log the value it will eventually emit.
 
-Subscribing to the Observable we created earlier would print:
+```javascript
+
+getData$.subscribe(data => {
+  // when the getData$ observable fires, log the data
+  console.log(data);
+});
```

Comment:
In the context of Angular, creating observables is probably an advanced topic since the Angular http service gives them to us.  Maybe we just say "assuming `getDataa$` is an observable we already have..."